internal/cmd/config_test.go

```diff
@@ -202,3 +206,199 @@ func TestDeriveAndValidateConfig(t *testing.T) {
+	memfs := fsext.NewMemMapFs()
+
+	conf := []byte(`{"iterations":1028,"cloud":{"field1":"testvalue"}}`)
+	defaultConfigPath := ".config/loadimpact/k6/config.json"
+	require.NoError(t, fsext.WriteFile(memfs, defaultConfigPath, conf, 0o644))
+
+	defaultFlags := state.GetDefaultFlags(".config")
+	gs := &state.GlobalState{
+		FS:           memfs,
+		Flags:        defaultFlags,
+		DefaultFlags: defaultFlags,
+	}
+	c, err := readDiskConfig(gs)
+	require.NoError(t, err)
+
+	assert.Equal(t, c.Iterations.Int64, int64(1028))
+	assert.JSONEq(t, `{"field1":"testvalue"}`, string(c.Cloud))
+}
+
+func TestReadDiskConfigCustomFilePath(t *testing.T) {
+	t.Parallel()
+	memfs := fsext.NewMemMapFs()
+
+	conf := []byte(`{"iterations":1028,"cloud":{"field1":"testvalue"}}`)
+	require.NoError(t, fsext.WriteFile(memfs, "custom-path/config.json", conf, 0o644))
+
+	defaultFlags := state.GetDefaultFlags(".config")
+	gs := &state.GlobalState{
+		FS:           memfs,
+		Flags:        defaultFlags,
+		DefaultFlags: defaultFlags,
+	}
+	gs.Flags.ConfigFilePath = "custom-path/config.json"
+
+	c, err := readDiskConfig(gs)
+	require.NoError(t, err)
+
+	assert.Equal(t, c.Iterations.Int64, int64(1028))
+	assert.JSONEq(t, `{"field1":"testvalue"}`, string(c.Cloud))
+}
+
+func TestReadDiskConfigNotFoundSilenced(t *testing.T) {
+	t.Parallel()
+	memfs := fsext.NewMemMapFs()
+
+	// Put the file into a different and unexpected directory
+	conf := []byte(`{"iterations":1028,"cloud":{"field1":"testvalue"}}`)
+	defaultConfigPath := ".config/unknown-folder/k6/config.json"
+	require.NoError(t, fsext.WriteFile(memfs, defaultConfigPath, conf, 0o644))
+
+	defaultFlags := state.GetDefaultFlags(".config")
+	gs := &state.GlobalState{
+		FS:           memfs,
+		Flags:        defaultFlags,
+		DefaultFlags: defaultFlags,
+	}
+	c, err := readDiskConfig(gs)
+	assert.NoError(t, err)
+	assert.Empty(t, c)
+}
+
+func TestReadDiskConfigNotJSONExtension(t *testing.T) {
+	t.Parallel()
+	memfs := fsext.NewMemMapFs()
+
+	conf := []byte(`{"iterations":1028,"cloud":{"field1":"testvalue"}}`)
+	require.NoError(t, fsext.WriteFile(memfs, "custom-path/config.txt", conf, 0o644))
+
+	defaultFlags := state.GetDefaultFlags(".config")
+	gs := &state.GlobalState{
+		FS:           memfs,
+		DefaultFlags: defaultFlags,
+		Flags:        defaultFlags,
+	}
+	gs.Flags.ConfigFilePath = "custom-path/config.txt"
+
+	c, err := readDiskConfig(gs)
+	require.NoError(t, err)
+
+	assert.Equal(t, c.Iterations.Int64, int64(1028))
+	assert.JSONEq(t, `{"field1":"testvalue"}`, string(c.Cloud))
+}
+
+func TestReadDiskConfigNotJSONContentError(t *testing.T) {
+	t.Parallel()
+	memfs := fsext.NewMemMapFs()
+
+	conf := []byte(`bad json format`)
+	defaultConfigPath := ".config/loadimpact/k6/config.json"
+	require.NoError(t, fsext.WriteFile(memfs, defaultConfigPath, conf, 0o644))
+
+	gs := &state.GlobalState{
+		FS:    memfs,
+		Flags: state.GetDefaultFlags(".config"),
+	}
+	_, err := readDiskConfig(gs)
+	require.ErrorContains(t, err, "couldn't parse")
```

Comment:
We can also compare the error type instead of string matching (since we wrap the error in the code), but it's also fine as it is:

```suggestion
	var serr *json.SyntaxError
	assert.ErrorAs(t, err, &jerr)
```